tests/test_filter.py

```diff
@@ -383,6 +405,24 @@ def test_application_to_dataset():
     )
     filtered_dataset = filter.apply(dataset, ["y", "x"])
 
+    # test iterated version of the Gaussian filter
+    filter_n2 = Filter(
+        filter_scale=4,
+        dx_min=1,
+        filter_shape=FilterShape.GAUSSIAN,
+        grid_type=GridType.REGULAR,
+        n_iterations=2,
+    )
+    filtered_dataset_n2 = filter_n2.apply(dataset, ["y", "x"])
+    # The following tests whether a relative error bound in L^2 holds.
+    # If the polynomial approximations are accurate to within 0.01,
+    # then the precise bound would have 0.0301 ** 2 instead of 0.04 ** 2.
+    # Bumped up because the default n_steps is not guaranteed to give
+    # error less than *exactly* 0.01; just quite close.
+    assert ((filtered_dataset.spatial - filtered_dataset_n2.spatial) ** 2).sum() < (
+        0.04 ** 2
+    ) * (filtered_dataset.spatial ** 2).sum()
+
     # Temporal variables should be unaffected because the filter is only
```

Comment:
Thanks for adding this test for filter splitting! You put the test inside of `test_application_to_dataset()`, where the latter test was specifically designed to check if feature #90 does the correct thing, i.e., whether the filter works on a`xarray.Dataset` (rather than just an `xarray.DataArray`). 

It would be cleaner to introduce a separate test for the filter splitting and then run the test on all implemented grids / Laplacians. Something like

```
def test_filter_splitting(grid_type_and_input_ds, filter_args):
    "Some description of the test"

    grid_type, da, grid_vars = grid_type_and_input_ds

    iterated_filter_args = filter_args.copy()
    iterated_filter_args["n_iterations"] = 2

    filter = Filter(grid_type=grid_type, grid_vars=grid_vars, **filter_args)
    iterated_filter = Filter(grid_type=grid_type, grid_vars=grid_vars, **iterated_filter_args)

    filtered = filter.apply(da, dims=["y", "x"])
    iteratively_filtered = iterated_filter.apply(da, dims=["y", "x"])

    # now your assert statement

```

Reply:
I updated the tests. I am a rank amateur at Python testing, so please don't hesitate to criticize this.